### PR TITLE
Port planning_pipeline to ROS 2

### PR DESCRIPTION
**Pull request by anasarrak**

### CODE REVIEW

**Comment by vmayoral** [comment]
Same as https://github.com/ros-planning/moveit2/pull/62#issuecomment-480720464, consider the changes please.

File: moveit_ros/planning/planning_pipeline/src/planning_pipeline.cpp [hunk 1]
```diff
@@ -92,7 +93,7 @@ void planning_pipeline::PlanningPipeline::configure()
   }
   catch (pluginlib::PluginlibException& ex)
   {
-    ROS_FATAL_STREAM("Exception while creating planning plugin loader " << ex.what());
+    RCLCPP_FATAL(node_->get_logger(),"Exception while creating planning plugin loader %s" , ex.what());
```
Comment (mlautman) [inline moveit_ros/planning/planning_pipeline/src/planning_pipeline.cpp]: do we have to convert to c_str() here?

File: moveit_ros/planning/planning_pipeline/src/planning_pipeline.cpp [hunk 2]
```diff
@@ -172,27 +173,27 @@ void planning_pipeline::PlanningPipeline::configure()
 void planning_pipeline::PlanningPipeline::displayComputedMotionPlans(bool flag)
 {
   if (display_computed_motion_plans_ && !flag)
-    display_path_publisher_.shutdown();
+    display_path_publisher_.reset();
   else if (!display_computed_motion_plans_ && flag)
-    display_path_publisher_ = nh_.advertise<moveit_msgs::msg::DisplayTrajectory>(DISPLAY_PATH_TOPIC, 10, true);
+    display_path_publisher_ = node_->create_publisher<moveit_msgs::msg::DisplayTrajectory>(DISPLAY_PATH_TOPIC);
```
Comment (mlautman) [inline moveit_ros/planning/planning_pipeline/src/planning_pipeline.cpp]: Please use the [QOS history depth parameter](https://github.com/ros2/rclcpp/blob/af9ae4a61c24ba207f3db6a32c41e0e52cb29cb9/rclcpp/include/rclcpp/node.hpp#L165) if not the full [rmw_qos_profile_t](https://github.com/ros2/rclcpp/blob/af9ae4a61c24ba207f3db6a32c41e0e52cb29cb9/rclcpp/include/rclcpp/node.hpp#L182)

File: moveit_ros/planning/planning_pipeline/src/planning_pipeline.cpp [hunk 3]
```diff
@@ -172,27 +173,27 @@ void planning_pipeline::PlanningPipeline::configure()
...
 }
 
 void planning_pipeline::PlanningPipeline::publishReceivedRequests(bool flag)
 {
   if (publish_received_requests_ && !flag)
-    received_request_publisher_.shutdown();
+    received_request_publisher_.reset();
   else if (!publish_received_requests_ && flag)
-    received_request_publisher_ = nh_.advertise<moveit_msgs::msg::MotionPlanRequest>(MOTION_PLAN_REQUEST_TOPIC, 10, true);
+    received_request_publisher_ = node_->create_publisher<moveit_msgs::msg::MotionPlanRequest>(MOTION_PLAN_REQUEST_TOPIC);
```
Comment (mlautman) [inline moveit_ros/planning/planning_pipeline/src/planning_pipeline.cpp]: Please use the [QOS history depth parameter](https://github.com/ros2/rclcpp/blob/af9ae4a61c24ba207f3db6a32c41e0e52cb29cb9/rclcpp/include/rclcpp/node.hpp#L165)

File: moveit_ros/planning/planning_pipeline/src/planning_pipeline.cpp [hunk 4]
```diff
@@ -172,27 +173,27 @@ void planning_pipeline::PlanningPipeline::configure()
...
 }
 
 void planning_pipeline::PlanningPipeline::checkSolutionPaths(bool flag)
 {
   if (check_solution_paths_ && !flag)
-    contacts_publisher_.shutdown();
+    contacts_publisher_.reset();
   else if (!check_solution_paths_ && flag)
-    contacts_publisher_ = nh_.advertise<visualization_msgs::MarkerArray>(MOTION_CONTACTS_TOPIC, 100, true);
+    contacts_publisher_ = node_->create_publisher<visualization_msgs::msg::MarkerArray>(MOTION_CONTACTS_TOPIC);
```
Comment (mlautman) [inline moveit_ros/planning/planning_pipeline/src/planning_pipeline.cpp]: Please use the [QOS history depth parameter](https://github.com/ros2/rclcpp/blob/af9ae4a61c24ba207f3db6a32c41e0e52cb29cb9/rclcpp/include/rclcpp/node.hpp#L165)

Comment (anasarrak) [reply moveit_ros/planning/planning_pipeline/src/planning_pipeline.cpp]: https://github.com/ros-planning/moveit2/pull/75/commits/bcbceba361836ef7bacb825fdb3784a7339a4a53

File: moveit_ros/planning/planning_pipeline/src/planning_pipeline.cpp (hunk 3 above)
Comment (anasarrak) [reply moveit_ros/planning/planning_pipeline/src/planning_pipeline.cpp]: https://github.com/ros-planning/moveit2/pull/75/commits/bcbceba361836ef7bacb825fdb3784a7339a4a53

File: moveit_ros/planning/planning_pipeline/src/planning_pipeline.cpp (hunk 2 above)
Comment (anasarrak) [reply moveit_ros/planning/planning_pipeline/src/planning_pipeline.cpp]: https://github.com/ros-planning/moveit2/pull/75/commits/bcbceba361836ef7bacb825fdb3784a7339a4a53

**Comment by anasarrak** [comment]
Also updated the moveit_ros_planning cmakelist https://github.com/ros-planning/moveit2/pull/75/commits/be0680512dd8bd1cd32e37f171ce4948f304bd0d, please @mlautman take a look

File: moveit_ros/planning/planning_pipeline/src/planning_pipeline.cpp [hunk 5]
```diff
@@ -49,18 +47,21 @@ const std::string planning_pipeline::PlanningPipeline::MOTION_PLAN_REQUEST_TOPIC
...
-                                                      const ros::NodeHandle& nh,
+                                                      const std::shared_ptr<rclcpp::Node> node,
                                                       const std::string& planner_plugin_param_name,
                                                       const std::string& adapter_plugins_param_name)
-  : nh_(nh), robot_model_(model)
+  : node_(node), robot_model_(model)
 {
+  auto planner_plugin_params = std::make_shared<rclcpp::SyncParametersClient>(node);
+
   std::string planner;
```
[the remark this answers is not in the document]
Comment (mlautman) [reply moveit_ros/planning/planning_pipeline/src/planning_pipeline.cpp]: @anasarrak I can merge this once you have addressed this

[the remark this answers is not in the document]
Comment (anasarrak) [reply moveit_ros/planning/planning_pipeline/src/planning_pipeline.cpp]: https://github.com/ros-planning/moveit2/pull/75/commits/0f64cb41e30a99cbbf95cbbcbfcbb0cb161df2d3

**Pull request merged**